webapp/src/js/services/report-view-model-generator.js

```diff
@@ -75,6 +76,11 @@ angular.module('inboxServices').factory('ReportViewModelGenerator',
       });
     };
 
+    const addContactUrl = (obj, model) => {
+      Object.assign(obj, { url: $state.href('contacts.detail',
+        { id: model.formatted.subject._id || model.formatted.fields.patient_uuid })});
```

Comment:
Why not just...

```js
obj.url = $state.href('contacts.detail', { id: model.formatted.subject._id || model.formatted.fields.patient_uuid });
```

Reply:
I have done a lot of clean up and now reduced the changes to only 2 lines code
`hasUrl: patient_fields.includes(field)` [Service]
`$scope.contactLink = (id) => $state.href('contacts.detail', { id: id });` [Controller]